src/main/java/com/redhat/digkins/DiggerClient.java

```diff
@@ -22,30 +22,71 @@
 
   public static final long DEFAULT_BUILD_TIMEOUT = 60 * 1000;
 
-  private final JenkinsServer jenkins;
+  private JenkinsServer jenkinsServer;
 
-  public DiggerClient(JenkinsAuth auth) throws URISyntaxException {
-    this.jenkins = new JenkinsServer(new URI(auth.getUrl()), auth.getUser(), auth.getPassword());
+  private CreateJobService createJobService;
+  private TriggerBuildService triggerBuildService;
+
+  private DiggerClient() {
   }
 
   /**
-   * Create client using provided url and credentials
+   * Create a client with defaults using provided url and credentials.
+   * <p>
+   * This client will use the defaults for the services. This is perfectly fine for majorith of the cases.
    *
    * @param url      Jenkins url
    * @param user     Jenkins user
    * @param password Jenkins password
    * @return client instance
    * @throws DiggerClientException if something goes wrong
    */
-  public static DiggerClient from(String url, String user, String password) throws DiggerClientException {
-    try {
-      JenkinsAuth jenkinsAuth = new JenkinsAuth(url, user, password);
-      return new DiggerClient(jenkinsAuth);
-    } catch (URISyntaxException e) {
-      throw new DiggerClientException("Invalid jenkins url format.");
+  public static DiggerClient createDefaultWithAuth(String url, String user, String password) throws DiggerClientException {
+    return DiggerClient.builder()
+      .createJobService(new CreateJobService())
+      .triggerBuildService(new TriggerBuildService(TriggerBuildService.DEFAULT_FIRST_CHECK_DELAY, TriggerBuildService.DEFAULT_POLL_PERIOD))
+      .withAuth(url, user, password)
+      .build();
+  }
+
+  public static DiggerClientBuilder builder() {
```

Comment:
Well, right..

I personally think that `DiggerClient.builder().foo().bar().build()` is more fluent than `new DiggerClientBuilder().foo().bar().build()`.
I am inspired by Guava in this idea.

I prefer keeping it for now, unless more objections arise.